league.js

```diff
@@ -0,0 +1,201 @@
+            } else {
+                console.log("Found " + pairings.length + " pairings for " + self.options.name);
+            }
+        });
+    },
+
+    //--------------------------------------------------------------------------
+    // Refreshes the latest roster information
+    //--------------------------------------------------------------------------
+    'refreshRosters': function(callback) {
+    },
+
+    //--------------------------------------------------------------------------
+    // Refreshes the latest mods information
+    //--------------------------------------------------------------------------
+    'refreshMods': function(callback) {
+    },
+
+    //--------------------------------------------------------------------------
+    // Figures out the current scheduling information for the round.
+    //--------------------------------------------------------------------------
+    'refreshCurrentRoundSchedules': function(callback) {
+        var query_options = {
+            'min-row': 1,
+            'max-row': 100,
+            'min-col': 1,
+            'max-col': 8,
+            'return-empty': true
+        }
+        var self = this;
+        spreadsheets.get_rows(
+            self.options.spreadsheet.service_account_auth,
+            self.options.spreadsheet.key,
+            query_options,
+            function(err, rows) {
+                if (err) { return callback(err, rows); }
+                var new_pairings = [];
+                rows.forEach(function(row) {
+                    if (!row['white'].value || !row['black'].value) { return; }
+                    var link = spreadsheets.parse_hyperlink(row['result'].formula || "");
+                    var date_string = row[self.options.spreadsheet.schedule_colname].value || '';
+                    date_string = date_string.trim()
+                    var date = moment.utc(
+                        moment.utc().year() + "/" + date_string,
+                        "YYYY/MM/DD @ HH:mm",
+                        true
+                    );
+                    if (!date.isValid()) {
+                        date = undefined;
+                    }
+                    new_pairings.push({
+                        white: self.canonical_username(row['white'].value),
+                        black: self.canonical_username(row['black'].value),
+                        result: link['text'],
+                        url: link['href'],
+                        scheduled_date: date
+                    });
+                });
+                self._pairings = new_pairings;
+                callback(undefined, self._pairings);
+            }
+        );
+    },
+
+    //--------------------------------------------------------------------------
+    // Finds the pairing for this current round given either a black or a white
+    // username.
+    //--------------------------------------------------------------------------
+    'findPairing': function(white, black) {
```

Comment:
This can be refactored no?  Something like:

    filterByColor(white, possibilities)
    filterByColor(black, possibilities)
 
    ...

    function filterByColor(color, possibilities) {
        if (color) {
              possibilities = _.filter(possibilities, function(item) {
                  return (
                      item.white.toLowerCase().includes(color) ||
                      item.black.toLowerCase().includes(color)
                  );
              });
          }
    }

If you get my meaning